pubtools/pulplib/_impl/model/maintenance.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import datetime
+import json
+import os
+
+import jsonschema
+
+from pubtools.pulplib._impl import compat_attr as attr
+from ..schema import load_schema
+from .common import InvalidDataException
+
+
+LOG = logging.getLogger("pubtools.pulplib")
+
+
+USER = os.environ.get("USER")
+HOSTNAME = os.environ.get("HOSTNAME")
+OWNER = "%s@%s" % (USER, HOSTNAME)
+
+
+def iso_time_now():
+    return datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceEntry(object):
+    """Details about the maintenance"""
+
+    id = attr.ib(default=None, type=str)
```

Comment:
Could we please call this "repository_id"?  I think bare "id" is risky - if maintenance entries are later linked to any other kind of entity which has an "id" - like an outage ID - then it becomes ambiguous.

Also, should it really have default=None?  I think this is probably the one attribute which is absolutely mandatory for every entry, and an entry without the repo ID would be unusable.

```suggestion
    repository_id = attr.ib(type=str)
```